blockchain/types/transaction_test.go

```diff
@@ -287,20 +412,167 @@ func TestIntrinsicGas(t *testing.T) {
 		data, err = hex.DecodeString(tc.inputString) // decode input string to hex data
 		assert.Equal(t, nil, err)
 
-		gas, err = IntrinsicGas(data, false, params.Rules{IsIstanbul: false})
+		gas, err = IntrinsicGas(data, nil, false, params.Rules{IsIstanbul: false})
 		assert.Equal(t, tc.expectGas1, gas)
 		assert.Equal(t, nil, err)
 
-		gas, err = IntrinsicGas(data, false, params.Rules{IsIstanbul: true})
+		gas, err = IntrinsicGas(data, nil, false, params.Rules{IsIstanbul: true})
 		assert.Equal(t, tc.expectGas2, gas)
 		assert.Equal(t, nil, err)
 
-		gas, err = IntrinsicGas(data, true, params.Rules{IsIstanbul: false})
+		gas, err = IntrinsicGas(data, nil, true, params.Rules{IsIstanbul: false})
 		assert.Equal(t, tc.expectGas3, gas)
 		assert.Equal(t, nil, err)
 
-		gas, err = IntrinsicGas(data, true, params.Rules{IsIstanbul: true})
+		gas, err = IntrinsicGas(data, nil, true, params.Rules{IsIstanbul: true})
 		assert.Equal(t, tc.expectGas4, gas)
 		assert.Equal(t, nil, err)
 	}
 }
+
+func TestIsFeeDelegation(t *testing.T) {
+	assert.False(t, TxTypeLegacyTransaction.IsFeeDelegatedTransaction())
```

Comment:
Maybe add for new type?

Reply:
This test function duplicated `blockchain/types/tx_internal_data_fee_ratio_test.go`.
I will remove this test.

Reply:
I removed it.